src/main/java/no/entur/uttu/stopplace/NetexPublicationDeliveryFileStopPlaceRegistry.java

```diff
@@ -84,4 +86,37 @@ public void init() {
   public Optional<StopPlace> getStopPlaceByQuayRef(String quayRef) {
     return Optional.ofNullable(stopPlaceByQuayRefIndex.get(quayRef));
   }
+
+  @Override
+  public List<StopPlace> getStopPlaces(List<StopPlaceFilter> filters) {
+    List<StopPlace> allStopPlaces = stopPlaceByQuayRefIndex
+      .values()
+      .stream()
+      .distinct()
+      .toList();
+    if (filters.isEmpty()) {
+      return allStopPlaces;
+    }
+
+    return allStopPlaces
+      .stream()
+      .filter(s -> isStopPlaceToBeIncluded(s, filters))
+      .toList();
+  }
+
+  private boolean isStopPlaceToBeIncluded(
+    StopPlace stopPlace,
+    List<StopPlaceFilter> filters
+  ) {
+    for (StopPlaceFilter f : filters) {
+      if (f instanceof TransportModeStopPlaceFilter transportModeStopPlaceFilter) {
```

Comment:
I suggest throw an error if filter is not implemented?

Reply:
Was just thinking if we still want to leave a possibility to query all 🤔

Reply:
Not sure what you mean by that. The filters list will only contain actual filters that api users want to use. So if you want all, then you provide no filters. If you provide a filter that is not implemented, it's an error that should be raised.

Reply:
Oh, sorry, I completely misunderstood! Yes, for sure, will add that 🙏

Reply:
Would `IllegalArgumentException` kind of exception be good for this case?